Production Environment (master branch)

lakshmantgld opened the pull request:
## Config.copy.yml
- Removed unwanted fields.

## sagawa/index.js
- removed productionURL

## sagawa/helpers
- removed sagawaEnv check.

## transaction/helpers
- removed squareEnv check.

## serverless.yml
- changed to config.prod.yml.
- added template variable to **stage**.

Commit:
modified config.yml to config.prod.yml & deleted unwanted envs

Commit:
removed the check on sagawaEnv

Commit:
removed productionUrl and the check on lambdaEnv

Commit:
removed the check on squareEnv

Commit:
removed the check on squareEnv

Commit:
added lambdaEnv to stage in serverless.yml

TobiahRex commented on serverless/db/mongo/models/sagawa/index.js:
```diff
@@ -509,12 +509,10 @@ export default (db) => {
       const token = JWT.sign(payload, jwtSecret);
-      const prodEnv = lambdaEnv === 'production';
-      const trackingLink = `${prodEnv ? `${productionUrl}/#!` : `${baseUrl}/#!`}/tracking?token=${token}`;
+      const trackingLink = `${baseUrl}/tracking?token=${token}`;
```
you need to add `/#!` so it should look like `${baseUrl}/#!/tracking?token=${token}`

lakshmantgld replied:
Nice Catch!!

lakshmantgld replied:
Fixed it!!

TobiahRex reviewed (changes requested):
There's a #!/ you left off for the token url.

Commit:
removed the unwanted envs

Commit:
added #! for the tracking link

Commit:
replicated the changes to server/

Commit:
hardcoded local mongo connection

Commit:
ignored the development version of .env

Commit:
modified the template file for .env by removing unwanted envs and adding new envs

Commit:
added different stages of .env file

Commit:
removed the check on NODE_ENV and added CLIENT_GRAPHQL_URL

Commit:
removed the check on NODE_ENV and exported configureStore.prod, since it is production branch

Commit:
removed the NODE_ENV check and replaced with just AUTH0_REDIRECT

Commit:
src/services/utils/squarePaymentForm.js

Commit:
removed the sandbox ens and check based on NODE_ENV

Commit:
removed the unwnated envs and added the new envs

lakshmantgld commented:
Introduced different environment in the **front-end:**
- Changed `API_GATEWAY_URL` to `CLIENT_GRAPHQL_URL` and changed the associated code in the `src/`
- Removed the `sanbox` envs and associated code.
- Removed unwanted envs and added new envs in `webpack.envs.js`.
- Updated the `.env_example` according to present env changes.
- Removed the `NODE_ENV` check in the redux store(`src/redux/store/index.js`) and exported based on the environment.

TobiahRex commented:
Thanks @lakshmantgld
